test/util/race.go

```diff
@@ -43,3 +45,10 @@ func getEnvInt(name string, fallback int) int {
 	}
 	return fallback
 }
+
+func SkipFlakyTestUnderRaceDetector(t *testing.T) {
```

Comment:
Can we add a comment here or maybe change the name so that it's clear new callers should not be added without careful consideration?

`SkipFlakyTestUnderRaceDetectorButYaBetterKnowWhatYouAreDoing` 😆

Reply:
I'm going to add an argument that should reference an open issue and at least match startswith or simple regex to minimally validate that it looks like an issue. Your naming suggestion reminds me of the time I had to reach for [this](https://sourcegraph.com/github.com/hashicorp/vagrant/-/blob/lib/vagrant/shared_helpers.rb?L69-74&subtree=true)...